### PR TITLE
[QS] Add monitoring counters for main loops in quorum store

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

Will help us understand if any quorum store loops are being saturated.

### CODE REVIEW

File: consensus/src/quorum_store/batch_generator.rs [hunk 1]
```diff
@@ -321,16 +323,16 @@ impl BatchGenerator {
...
                     match next_proof {
                         Ok(proof) => self.handle_completed_proof(proof).await,
                         Err(_) => {
                             debug!("QS: proof oneshot dropped");
                         }
                     }
-                },
-                Some(cmd) = cmd_rx.recv() => {
+                }),
+                Some(cmd) = cmd_rx.recv() =>monitor!("batch_generator_handle_command", {
```
Comment (zekun000) [inline consensus/src/quorum_store/batch_generator.rs:335]: a space between => and m

**Review by zekun000 — COMMENTED** [review]
there's a few other components I think, batch_coordinator, network_listener

**Comment by sitalkedia** [comment]
> there's a few other components I think, batch_coordinator, network_listener

Added for network_listener, for `batch_coordinator`, its a bit convoluted, as there are multiple instances. Will add it as a follow up.

File: consensus/src/quorum_store/quorum_store_coordinator.rs [hunk 2]
```diff
@@ -54,108 +55,112 @@ impl QuorumStoreCoordinator {
...
-                        .expect("Failed to send to BatchGenerator");
-                },
-                CoordinatorCommand::Shutdown(ack_tx) => {
-                    // Note: Shutdown is done from the back of the quorum store pipeline to the
-                    // front, so senders are always shutdown before receivers. This avoids sending
-                    // messages through closed channels during shutdown.
-                    // Oneshots that send data in the reverse order of the pipeline must assume that
-                    // the receiver could be unavailable during shutdown, and resolve this without
-                    // panicking.
+            monitor!("batch_store_coordinator_loop", {
```
Comment (zekun000) [inline consensus/src/quorum_store/quorum_store_coordinator.rs:58]: this is not batch_store

Comment (sitalkedia) [reply consensus/src/quorum_store/quorum_store_coordinator.rs]: Fixed.

**Review by ibalajiarun — APPROVED** [review]

**Review by zekun000 — APPROVED** [review]

**Pull request merged**